catalogs/catalog-lakehouse-paimon/src/main/java/org/apache/gravitino/catalog/lakehouse/paimon/GravitinoPaimonTable.java

```diff
@@ -51,7 +64,25 @@ protected TableOperations newOps() {
    * @return The converted Paimon table.
    */
   public Schema toPaimonTableSchema() {
-    Schema.Builder builder = Schema.newBuilder().comment(comment).options(properties);
+    Schema.Builder builder = Schema.newBuilder().comment(comment);
+    if (properties == null) {
+      properties = Maps.newHashMap();
+    }
+    if (partitioning == null) {
+      partitioning = EMPTY_PARTITIONING;
+    }
+
+    Map<String, String> normalizedProperties = new HashMap<>(properties);
+    normalizedProperties.remove(PRIMARY_KEY_IDENTIFIER);
```

Comment:
Gravitino has `PRIMARY_KEY` index, could paimon catalog use it?

Reply:
imo, the Gravitino `Index` should be designed for `RDBMS`, am i right?
although paimon's primary key is also unique, there are some differences between it and `RDBMS's` primary key, for example: 
1.  Index in `RDBMS` is globally ordered, but paimon's primary key is ordered  in bucket level.
2.  insert a duplicate record will report an error in RDBMS, but now in paimon, if there is a duplicate record, the internal will be implicit deduplication for insert statement.
3. paimon has not supported the clear concept of column Index so far.  If paimon supports it, we can support it later.

Reply:
Hi @FANNG1 do you have further comments about this?

Reply:
@jerryshao you thought?

Reply:
After discuss with @FANNG1 privately, I think it is suitable to use `PRIMARY_INDEX` than a property to define a key here. Though the behavior of Paimon is slightly different from RDBMS, but it is still some sorts of primary key, right? In Gravitino, we only semantically unify them, not in behavior, each implementation can have it's own explanation, like what we defined for types, transforms, distributions, we only define them semantically, but we don't restrict the behavior, each implementation may be different. So this also applies to here.

Still, we can extend the current PRIMARY_INDEX definition to better support Paimon, WDYT? @caican00

Reply:
> but it is still some sorts of primary key, right? 
 yeah.

got it, and it sounds reasonable. I'll use `Index` instead

Reply:
fixed.